### ISSUE TITLE
Error: Demographic Citizen would reference itself with this new definition.

### ISSUE THREAD

**Issue by Deantwo** [StrangeLoopGames/EcoIssues#12499]
Trying to redefine the Citizen demographic on the official Sea Otter server.
![image](https://user-images.githubusercontent.com/2676134/59147076-53f4f280-89f7-11e9-8327-d52b421a8f1a.png)
The website is extremely slow to create the law, and when i am done crafting it I get this nonsense error message.
```
Error: Demographic Citizen would reference itself with this new definition.
```

Related: #9859

I guess this issue is gonna be resolved in v0.9.0.0, but figured I'd report it just in case. It could also just be the official Sea Otter server having issues. @D3nnis3n maybe you can pull a copy of the Sea Otter server save, so we can see if it can be reproduced.

**Comment by Deantwo** [comment]
This can probably be closed with v9.0, as editing laws and demographics work totally differently.

**Comment by WeaselDog** [comment]
Fixed with 9.0
